modules/dag-l0/src/main/scala/org/tessellation/dag/l0/domain/snapshot/programs/GlobalSnapshotEventCutter.scala

```diff
@@ -0,0 +1,144 @@
+package org.tessellation.dag.l0.domain.snapshot.programs
+
+import cats.MonadThrow
+import cats.effect.Async
+import cats.syntax.applicative._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.foldable._
+import cats.syntax.functor._
+import cats.syntax.option._
+import cats.syntax.traverse._
+import cats.syntax.unorderedFoldable._
+
+import org.tessellation.dag.l0.infrastructure.snapshot.{DAGEvent, GlobalSnapshotEvent, StateChannelEvent}
+import org.tessellation.ext.kryo.RefinedSerializer
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.node.shared.domain.event.EventCutter
+import org.tessellation.schema.Block
+import org.tessellation.schema.gossip.Ordinal
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.security.hash._
+import org.tessellation.security.{Hashed, Hasher}
+
+import eu.timepit.refined.types.numeric.PosInt
+
+object GlobalSnapshotEventCutter {
+  type GlobalPeerIdEventMap = Map[PeerId, List[(Ordinal, GlobalSnapshotEvent)]]
+
+  def make[F[_]: Async: KryoSerializer: Hasher](maxBinarySizeBytes: PosInt): EventCutter[F, PeerId, Ordinal, GlobalSnapshotEvent] =
+    new EventCutter[F, PeerId, Ordinal, GlobalSnapshotEvent] {
+
+      def cut(events: GlobalPeerIdEventMap): F[GlobalPeerIdEventMap] =
```

Comment:
I think this function discards the cut events and unless it's intentionally introduced in this PR we should change it as we have a logic in other places returning such events back to the pool/cache.

Reply:
@ryle-goehausen Can you answer this? Do we need to retain the cut events in a separate collection?